api/server/extension_points.go

```diff
@@ -40,31 +40,27 @@ func (s *Server) apiRouteHandlerWrapperFunc(apiHandler fnext.ApiRouteHandler) gi
 	return func(c *gin.Context) {
 		context := c.Request.Context()
 		// get the app
-		appName := c.Param(api.CApp)
-		app, err := s.datastore.GetApp(context, appName)
+		appID := c.MustGet(api.AppID).(string)
+		// get the route TODO
+		routePath := "/" + c.Param(api.CRoute)
+		route, err := s.datastore.GetRoute(context, appID, routePath)
 		if err != nil {
 			handleErrorResponse(c, err)
 			c.Abort()
 			return
 		}
-		if app == nil {
-			handleErrorResponse(c, models.ErrAppsNotFound)
+		if route == nil {
+			handleErrorResponse(c, models.ErrRoutesNotFound)
 			c.Abort()
 			return
 		}
-		// get the route TODO
-		routePath := "/" + c.Param(api.CRoute)
-		route, err := s.datastore.GetRoute(context, appName, routePath)
+
+		app, err := s.datastore.GetAppByID(context, appID)
 		if err != nil {
 			handleErrorResponse(c, err)
 			c.Abort()
 			return
 		}
-		if route == nil {
```

Comment:
probably still need the app check just in case I guess

Reply:
fixed